src/ReachSets/reach.jl

```diff
@@ -211,6 +214,130 @@ function reach(S::AbstractSystem,
+               kwargs...)::Vector{<:LazySet}
+               # unpack arguments
+               kwargs_dict = Dict(kwargs)
+
+
+
+               # list containing the arguments passed to any reachability function
+               args = []
+               waiting_list = []
+               # coefficients matrix
+               #TODO get start state. For now we assume that it is the first location
+               cur_loc_id = 1
+               cur_loc = HS.modes[init_loc_id]
+
+               #TODO push_initial_set
+               push!((cur_loc[1], X0), waiting_list)
+               i = 0
+               while (!isempty(waiting_list) and i < 15) #TODO add variable for max iteration number
+
+                   S = ContinuousSystem(cur_loc.A, X0, cur_loc.U)
+
+                   Rsets = reach(
+                        S,
+                        kwargs_dict[:N];
+                        algorithm=kwargs_dict[:algorithm],
+                        ε_init=kwargs_dict[:ε_init],
+                        set_type_init=kwargs_dict[:set_type_init],
+                        ε_iter=kwargs_dict[:ε_iter],
+                        set_type_iter=kwargs_dict[:set_type_iter],
+                        assume_sparse=kwargs_dict[:assume_sparse],
+                        assume_homogeneous=kwargs_dict[:assume_homogeneous],
+                        lazy_X0=kwargs_dict[:lazy_X0],
+                        blocks=kwargs_dict[:blocks],
+                        partition=kwargs_dict[:partition],
+                        block_types_init=kwargs_dict[:block_types_init],
+                        block_types_iter=kwargs_dict[:block_types_iter],
+                        template_directions_init=kwargs_dict[:template_directions_init],
+                        template_directions_iter=kwargs_dict[:template_directions_iter],
+                        lazy_inputs_interval=kwargs_dict[:lazy_inputs_interval],
+                        output_function=kwargs_dict[:output_function]
+                        )
+                       for (j, tj) in enumerate(transitions(HS)) #TODO optimize it or add method for outgoing transitions to SX
+
+                            if source(HS, tj) == cur_loc_id
+                                destination_loc = HS.modes[target(HS, tj)]
+                                source_invariant, target_invariant = cur_loc[2], destination_loc[2]
+                                #guard =
+                                ```
+                                get transition_info:
+                                Guard
+                                Destination_location
+                                # check intersection G_I^+_i^-
+                                if (intersection != empty)
+                                    push!(waiting_list, (inter_set, loc_id))
+                                ```
+                            end
+                        end
+                    cur_loc, X0 = pop!(waiting_list)
+                    i += 1
```

Comment:
I think the `pop!` should happen at the start of the loop.

Reply:
yeah, good catch, I thought that it could be a clash, but I was wrong.